### PR TITLE
SAAS-182: init database if empty before starting jahia

### PR DESCRIPTION
**Pull request by Ronan-Gorain**
https://jira.jahia.org/browse/SAAS-182

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,5 +1,18 @@
 #!/bin/bash
 
+function check_db_access {
+    for n in {1..667}; do
+        echo -n "Testing network database access on host $DB_HOST (test $n)... "
+        [ $n -gt 666 ] && echo "We are doomed !" && exit 1
```
Comment (laurentfufu) [inline entrypoint.sh:6]:
```suggestion
        [ $n -gt 666 ] && echo "We are doomed !" && exit 1
        echo -n "Testing network database access on host $DB_HOST (test $n)... "
```

File: entrypoint.sh [hunk 2]
```diff
@@ -72,29 +85,47 @@ echo "$SUPER_USER_PASSWORD" > $FACTORY_DATA/root.pwd
 
 
 case "$DBMS_TYPE" in
-    "mariadb") 
+    "mariadb")
         DB_PORT="3306"
+        check_db_access
+
+        db_exist=$(mysql -h $DB_HOST -u $DB_USER -p$DB_PASS -e "show databases like 'coucou'")
```
Comment (laurentfufu) [inline entrypoint.sh:92]: not sure about _coucou_ database

File: entrypoint.sh [hunk 3]
```diff
@@ -72,29 +85,47 @@ echo "$SUPER_USER_PASSWORD" > $FACTORY_DATA/root.pwd
...
+        if [ "$db_exist" == "" ]; then
+            echo "Database doesn't exist. Trying to create it..."
+            mysql -h $DB_HOST -u $DB_USER -p$DB_PASS -e "create database $DB_NAME"
+        fi
```
Comment (laurentfufu) [inline entrypoint.sh:96]:
```suggestion
        if [ "$db_exist" == "" ]; then
            echo "Database doesn't exist. Trying to create it..."
            mysql -h $DB_HOST -u $DB_USER -p$DB_PASS -e "create database $DB_NAME CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;"
        fi
```

have to be sure that the database is created in order to continue or break too

File: entrypoint.sh [hunk 4]
```diff
@@ -72,29 +85,47 @@ echo "$SUPER_USER_PASSWORD" > $FACTORY_DATA/root.pwd
...
+            mysql -h $DB_HOST -u $DB_USER -p$DB_PASS $DB_NAME <<< $(cat /data/digital-factory-data/db/sql/schema/mysql/01*)
+        fi
+
         testdb_result="$(mysql -u $DB_USER -p$DB_PASS -h $DB_HOST -D $DB_NAME -e "select count(REVISION_ID) from JR_J_LOCAL_REVISIONS;" -s)"
         ;;
     "postgresql")
         DB_PORT="5432"
+        check_db_access
+
+        db_exists=$(PGPASSWORD=$DB_PASS psql -t -h $DB_HOST -U $DB_USER -d postgres -c "SELECT datname FROM pg_database where datname='$DB_NAME';")
```
Comment (laurentfufu) [inline entrypoint.sh:110]:
```suggestion
        db_exists=$(PGPASSWORD=$DB_PASS psql -t -h $DB_HOST -U $DB_USER  -c "SELECT datname FROM pg_database where datname='$DB_NAME';")
```

File: entrypoint.sh [hunk 5]
```diff
@@ -72,29 +85,47 @@ echo "$SUPER_USER_PASSWORD" > $FACTORY_DATA/root.pwd
...
+        if [ "$db_exists" == "" ]; then
+            echo "Database doesn't exist. Trying to create it..."
+            PGPASSWORD=$DB_PASS psql -h $DB_HOST -U $DB_USER -d postgres -c "create database $DB_NAME"
```
Comment (laurentfufu) [inline entrypoint.sh:113]:
```suggestion
            PGPASSWORD=$DB_PASS psql -h $DB_HOST -U $DB_USER  -c "create database $DB_NAME"
```


have to be sure that the database is created in order to continue or break too

Comment (Ronan-Gorain) [reply entrypoint.sh:117]: Could find a solution to remove  this " -d postgres"

**Pull request merged**